hadoop-hdfs-project/hadoop-hdfs-rbf/src/main/java/org/apache/hadoop/hdfs/server/federation/store/driver/impl/StateStoreFileBaseImpl.java

```diff
@@ -366,7 +375,7 @@ public <T extends BaseRecord> boolean putAll(
         }
       }
       // Commit
-      if (!rename(recordPathTemp, recordPath)) {
+      if (success && !rename(recordPathTemp, recordPath)) {
```

Comment:
This logic doesn't seem right.`success` is scoped to the function and this rename is within the for loop. As soon as any write fails we will never commit any changes; if that'd the case we should likely just `break`. 

So either we want to not commit any changes after the first failure, in which case we should break or we need a different variable.

Reply:
Good catch @mkuchenbecker, thanks. I've added a new variable. I prefer adding a new variable versus breaking out of the loop. That way we only skip renaming files with erroneous writes, and ordering of records isn't considered.